src/nspawn/nspawn.c

```diff
@@ -3206,6 +3206,17 @@ static int outer_child(
         return 0;
 }
 
+static int on_sigchld(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
+        pid_t pid;
+
+        pid = PTR_TO_UINT32(userdata);
+        if (pid == si->ssi_pid) {
```

Comment:
Actually the latter happens because nspawn does "prctl(PR_SET_CHILD_SUBREAPER, 1)". It doesn't happen under a CLONE_NEWPID container because the kernel then re-parents orphans to the root process of the Pid NS.